File: leetcode/js/remove-all-occurrences-of-a-substring.js
/**
 * @param {string} s
 * @param {string} part
 * @return {string}
 */
var removeOccurrences = function (s, part) {
  let result = s;

  while (true) {
    const replaced = result.replace(part, "");

    if (replaced.length === result.length) {
      break;
    }

    result = replaced;
  }

  return result;
};

File: leetcode/rust/remove-all-occurrences-of-a-substring.rs
impl Solution {
    pub fn remove_occurrences(s: String, part: String) -> String {
        let mut result = s;

        loop {
            let replaced = result.replacen(&part, "", 1);

            if result.len() == replaced.len() {
                break;
            }

            result = replaced;
        }

        result
    }
}
